api/user/user.model.js

```diff
@@ -1,8 +1,28 @@
 const mongoose = require('mongoose');
+const bcrypt = require('bcrypt-nodejs');
+const Promise = require('bluebird');
 
-const userSchema = new mongoose.Schema({
+
+const userSchema = mongoose.Schema({
   username: { type: String, required: true, index: { unique: true } },
   password: { type: String, required: true },
 });
 
-module.exports = mongoose.model('User', userSchema);
+const User = mongoose.model('User', userSchema);
+
+User.comparePassword = (candidatePassword, savedPassword) => {
+  const compare = Promise.promisify(bcrypt.compare);
```

Comment:
I'm wondering if these's promisification of the helper functions should be outside the function bodies. That way they aren't promisified every time the function is called.